bundles/binding/org.openhab.binding.plclogo/src/main/java/org/openhab/binding/plclogo/internal/PLCLogoBinding.java

```diff
@@ -305,32 +307,38 @@ protected void internalReceiveCommand(String itemName, Command command) {
                 logger.debug("No S7client for controller {} found", config.getController());
                 continue;
             }
-            
+
             lock.lock();
-            int result = ReadLogoDBArea(LogoS7Client, controller.getMemorySize());
+
+            byte buf[] = new byte[2];
+            int size = wr.isDigital() ? 1 : 2;
+            int result = LogoS7Client.ReadArea(S7.S7AreaDB, 1, address, size, buf);
+            logger.debug("Read word from logo memory: at {} {} bytes, result = {}", address, size, result);
             if (result == 0) {
+
+                if (bit >= 0) {
```

Comment:
I would omit this if() {} else {} block. and place debug output into "set" - Section.
For word case just bevore S7.SetWordAt(...) Reason: Performance.